api/api/secret_storages_api.go

```diff
@@ -0,0 +1,205 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/caraml-dev/mlp/api/log"
+	"github.com/caraml-dev/mlp/api/models"
+)
+
+type SecretStoragesController struct {
+	*AppContext
+}
+
+// ListSecretStorage lists all secret storage for a project including the global secret storage
+func (c *SecretStoragesController) ListSecretStorage(_ *http.Request,
+	vars map[string]string,
+	_ interface{}) *Response {
+
+	projectID, _ := models.ParseID(vars["project_id"])
+	if projectID <= 0 {
+		log.Errorf("project_id is not valid: %d", projectID)
+		return BadRequest("project_id is not valid")
+	}
+
+	_, err := c.ProjectsService.FindByID(projectID)
+	if err != nil {
+		log.Errorf("error fetching project with ID: %d", projectID)
+		return FromError(err)
+	}
+
+	secretStorages, err := c.SecretStorageService.List(projectID)
+	if err != nil {
+		log.Errorf("error fetching secret storages for project with ID: %d", projectID)
+		return FromError(err)
+	}
+
+	return Ok(secretStorages)
+}
+
+// GetSecretStorage gets a secret storage given specified project_id and secret_storage_id
+func (c *SecretStoragesController) GetSecretStorage(_ *http.Request,
+	vars map[string]string,
+	_ interface{}) *Response {
+
+	projectID, _ := models.ParseID(vars["project_id"])
+	secretStorageID, _ := models.ParseID(vars["secret_storage_id"])
+	if projectID <= 0 || secretStorageID <= 0 {
```

Comment:
does this cause specific error that has to be handled otherwise compared to `secretStorage, err := c.SecretStorageService.FindByID(secretStorageID)`

Reply:
Yeah, it will throw `not found` if we don't have this guard.